cmake/External/nccl.cmake

```
@@ -30,20 +31,35 @@ if(NOT __NCCL_INCLUDED)
         "CUDA_HOME=${CUDA_TOOLKIT_ROOT_DIR}"
         "NVCC=${CUDA_NVCC_EXECUTABLE}"
         "NVCC_GENCODE=${NVCC_GENCODE}"
-        "BUILDDIR=${CMAKE_CURRENT_BINARY_DIR}/nccl"
+        "BUILDDIR=${__NCCL_BUILD_DIR}"
         "VERBOSE=0"
         "-j"
-      BUILD_BYPRODUCTS "${CMAKE_CURRENT_BINARY_DIR}/nccl/lib/libnccl_static.a"
+        BUILD_BYPRODUCTS "${__NCCL_BUILD_DIR}/lib/libnccl_static.a"
       INSTALL_COMMAND ""
       )
 
+    add_custom_command(
+      OUTPUT "${__NCCL_BUILD_DIR}/lib/libnccl_slim_static.a"
+      DEPENDS nccl_external
+      COMMAND "${CMAKE_COMMAND}" -E make_directory "${__NCCL_BUILD_DIR}/objects"
+      COMMAND pushd objects
```

Appears as though `pushd` doesn't exist in our images `sh` shell, at least for gcc 5.4.
```suggestion
      COMMAND cd objects
```

---

Updated